File: iOS/FlowWalletKit/Sources/Account/VM/FlowVM.swift
//
//  File.swift
//
//
//  Created by Hao Fu on 27/8/2024.
//

import Foundation
import Flow

public enum FlowVM {
    case EVM
}

public protocol FlowVMProtocol {
    associatedtype VMAddress
    
    var vm: FlowVM { get }
    var address: VMAddress { get }
    var chainID: Flow.ChainID { get }
}

File: Android/wallet/src/main/java/com/flow/wallet/account/vm/FlowVM.kt
package com.flow.wallet.account.vm
import org.onflow.flow.ChainId

/**
 * Enum representing different virtual machine types supported by Flow
 */
enum class FlowVM {
    EVM
}

/**
 * Interface for virtual machine protocol implementations
 */
interface FlowVMProtocol<VMAddress> {
    val vm: FlowVM
    val address: VMAddress
    val chainID: ChainId
}